sources/org.osbuild.ostree

```diff
@@ -43,6 +44,19 @@ SCHEMA = """
                   "type": "string",
                   "description": "GPG keys to verify the commits"
                 }
+              },
+              "secrets": {
+                "type": "object",
+                "additionalProperties": false,
+                "required": [
+                  "name"
+                ],
+                "properties": {
+                  "name": {
+                    "type": "string",
+                    "description": "Name of the secrets provider."
+                  }
+                }
```

Comment:
I am struggling to understand what this configuration does. I assume it makes `ostree` use a specific client-certificate/key when interacting with the remote that is configured? At least that's what I get from the implementation, but I feel like this schema is misleading in that case. `"secrets": { "name": "foobar" }` seems rather unspecific. Why not explicitly use: `"client-cert-provider": "foobar"`?

You also add an indirection by using the `name` key inside of the `secrets` object. Do you expect other keys to be added as well? If yes, which?

Lastly, I think from the configuration+documentation it is not clear that this specifies the name of the secret-provider. It is neither a path, nor an inline secret, but the name of a hard-coded provider that we support. Maybe we can turn this into a `one-of` entry and make the parser fail if you specify a provider we do not support? This would be forward-compatible to a free-form entry as you suggest, in case we extend support for it.

Reply:
I don't disagree, but I just copied this from the org.osbuild.curl source. If we want to make these more stringent I'm happy to do that.

Reply:
You are right. *sigh*
Lets defer to later then.